casper/src/test/scala/coop/rchain/casper/api/BlockQueryResponseAPITest.scala

```diff
@@ -59,10 +59,11 @@ class BlockQueryResponseAPITest
       setValidator = sender.some,
       setDeploys = randomDeploys.some,
       setJustifications = List(Justification(bondsValidator.validator, genesisBlock.blockHash)).some,
+      setParentsHashList = List(genesisBlock.blockHash).some,
```

Comment:
This test has been failing on `dag.unsafeLookup` method because default generated parent that is not in dagstore has been used. 
In previous code this was mitigated by how code is has been written
```scala
      private def computeMainParentWeightMap(
          blockDag: BlockDagRepresentation[F],
          candidateMetadata: BlockMetadata
      ): F[Map[BlockHash, Long]] =
        candidateMetadata.parents.headOption match {
          case Some(parent) =>
            blockDag.lookup(parent).map(_.fold(candidateMetadata.weightMap)(_.weightMap))
          case None => candidateMetadata.weightMap.pure[F]
        }
```

But in general - if we want to know fault tolerance of some block, I believe it's safe to assume (and we ned to force that) that if some message has main parent, that main parent has to be in dagstore. Otherwise throwing error is reasonable.

Fault tolerance is adjusted because target message is tip, and previously it had fault tolerance `+1` because default parent has been used.